### PR TITLE
feat: add dynamic fees math

### PR DESCRIPTION
**Pull request by enthusiastmartin**

### CODE REVIEW

**Review by Roznovjak — APPROVED** [review]
LGTM but I'm not familiar with the math used here so I couldn't verify it.

File: src/dynamic_fees/math.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+use crate::dynamic_fees::types::NetVolumeDirection::{InOut, OutIn};
+use crate::dynamic_fees::types::{FeeParams, NetVolumeDirection, OracleEntry};
```
Comment (apopiak) [inline src/dynamic_fees/math.rs]: why not `use` with `super::types`?

File: src/dynamic_fees/types.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/// Oracle entry data for an asset, providing amount in and out and total liquidity of an asset.
+#[derive(Debug, Clone)]
+pub struct OracleEntry {
+    pub amount_in: Balance,
+    pub amount_out: Balance,
+    pub liquidity: Balance,
+}
+
+impl OracleEntry {
+    pub(super) fn net_volume(&self, direction: NetVolumeDirection) -> (Balance, bool) {
```
Comment (apopiak) [inline src/dynamic_fees/types.rs:23]: docs, please (specifically describe what the `bool` represents (it's whether the return value is negative, correct?)

any reason to restrict usage to this module?

**Review by apopiak — CHANGES_REQUESTED** [review]
looks fine, but where are the tests?

File: src/dynamic_fees/types.rs (hunk 2 above)
Comment (enthusiastmartin) [reply src/dynamic_fees/types.rs:23]: this is internal helper.  no need to expose it.

**Pull request merged**